crypto/pkcs7/pkcs7.c

```diff
@@ -608,3 +609,480 @@ void PKCS7_RECIP_INFO_get0_alg(PKCS7_RECIP_INFO *ri, X509_ALGOR **penc) {
+        goto err;
+      }
+      break;
+    case NID_pkcs7_enveloped:
+      rsk = p7->d.enveloped->recipientinfo;
+      xalg = p7->d.enveloped->enc_data->algorithm;
+      evp_cipher = p7->d.enveloped->enc_data->cipher;
+      if (evp_cipher == NULL) {
+        OPENSSL_PUT_ERROR(PKCS7, PKCS7_R_CIPHER_NOT_INITIALIZED);
+        goto err;
+      }
+      break;
+    case NID_pkcs7_digest:
+      content = PKCS7_get_octet_string(p7->d.digest->contents);
+      if (!pkcs7_bio_add_digest(&out, p7->d.digest->digest_alg)) {
+        goto err;
+      }
+      break;
+    case NID_pkcs7_data:
+      break;
+    default:
+      OPENSSL_PUT_ERROR(PKCS7, PKCS7_R_UNSUPPORTED_CONTENT_TYPE);
+      goto err;
+  }
+
+
+  for (size_t i = 0; i < sk_X509_ALGOR_num(md_sk); i++) {
+    if (!pkcs7_bio_add_digest(&out, sk_X509_ALGOR_value(md_sk, i))) {
+      goto err;
+    }
+  }
+
+  if (evp_cipher != NULL) {
+    unsigned char key[EVP_MAX_KEY_LENGTH];
+    unsigned char iv[EVP_MAX_IV_LENGTH];
+    int keylen, ivlen;
+    EVP_CIPHER_CTX *ctx;
+
+    if ((btmp = BIO_new(BIO_f_cipher())) == NULL) {
+      OPENSSL_PUT_ERROR(PKCS7, ERR_R_BIO_LIB);
+      goto err;
+    }
+    if (!BIO_get_cipher_ctx(btmp, &ctx)) {
+      OPENSSL_PUT_ERROR(PKCS7, ERR_R_BIO_LIB);
+      goto err;
+    }
+    keylen = EVP_CIPHER_key_length(evp_cipher);
+    ivlen = EVP_CIPHER_iv_length(evp_cipher);
+    ASN1_OBJECT_free(xalg->algorithm);
+    xalg->algorithm = OBJ_nid2obj(EVP_CIPHER_nid(evp_cipher));
+    if (ivlen > 0) {
+      RAND_bytes(iv, ivlen);
+    }
+    if (keylen > 0) {
+      RAND_bytes(key, keylen);
+    }
+
+    if (EVP_CipherInit_ex(ctx, evp_cipher, NULL, key, iv, /*enc*/ 1) <= 0) {
+      goto err;
+    }
+
+    if (ivlen > 0) {
+      ASN1_TYPE_free(xalg->parameter);
+      xalg->parameter = ASN1_TYPE_new();
+      if (xalg->parameter == NULL) {
+        goto err;
+      }
+      xalg->parameter->type = V_ASN1_OCTET_STRING;
+      xalg->parameter->value.octet_string = ASN1_OCTET_STRING_new();
+      // Set |p7|'s parameter value to the IV
+      if (!ASN1_OCTET_STRING_set(xalg->parameter->value.octet_string, iv,
+                                 ivlen)) {
+        goto err;
+      }
+    }
+
+    for (size_t i = 0; i < sk_PKCS7_RECIP_INFO_num(rsk); i++) {
+      ri = sk_PKCS7_RECIP_INFO_value(rsk, i);
+      if (pkcs7_encode_rinfo(ri, key, keylen) <= 0) {
+        goto err;
+      }
+    }
+    OPENSSL_cleanse(key, keylen);
+
+    if (out == NULL) {
+      out = btmp;
+    } else {
+      BIO_push(out, btmp);
+    }
+    btmp = NULL;
+  }
+
+  if (bio == NULL) {
+    bio = BIO_new(BIO_s_mem());
+    if (bio == NULL) {
+      goto err;
+    }
+    BIO_set_mem_eof_return(bio, /*eof_value*/ 0);
+    // clang-format off
+OPENSSL_BEGIN_ALLOW_DEPRECATED
+    // clang-format on
+    if (!PKCS7_is_detached(p7) && content && content->length > 0) {
+      // clang-format off
+OPENSSL_END_ALLOW_DEPRECATED
+      // clang-format on
```

Comment:
I experimented with a few options in the `.clang-format` file to see if there's some way to get it to do what we want.  I didn't have any luck.

Needing to turn on/off clang-format each time might be worse than just letting the macros be indented.  So I'm fine if you prefer to just let them be indented. Either way.

Reply:
ah, bummer. i'll re-indent them in the next PR.